server/src/services/users.service.ts

```diff
@@ -147,24 +147,21 @@ export class UsersService {
 
     const hashedPassword = password ? bcrypt.hashSync(password, 10) : undefined;
 
-    const updateableParams = {
+    const updatableParams = {
       forgotPasswordToken,
       firstName,
       lastName,
       password: hashedPassword,
     };
 
     // removing keys with undefined values
-    cleanObject(updateableParams);
+    cleanObject(updatableParams);
 
-    let user: User;
+    await this.updateUser(userId, updatableParams);
```

Comment:
why remove it outside db transaction block? 🤔

Reply:
to reuse the `updateUser`  function

Reply:
If we are doing multiple updates, better to group it in a transaction to avoid partial updates being done. Can modify the function to recieve the entity manager inorder to do so.